pkg/k8s/metadata.go

```diff
@@ -0,0 +1,102 @@
+package k8s
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"encoding/json"
+	"github.com/feedhenry/mcp-standalone/pkg/mobile"
+	"github.com/pkg/errors"
+)
+
+type Metadata struct {
+	Issuer                        string   `json:"issuer"`
+	AuthorizationEndpoint         string   `json:"authorization_endpoint"`
+	TokenEndpoint                 string   `json:"token_endpoint"`
+	ScopesSupported               []string `json:"scopes_supported"`
+	ResponseTypesSupported        []string `json:"response_types_supported"`
+	GrantTypesSupported           []string `json:"grant_types_supported"`
+	CodeChallengeMethodsSupported []string `json:"code_challenge_methods_supported"`
+}
+
+func GetMetadata(k8shost string, requester mobile.ExternalHTTPRequester) (*Metadata, error) {
+	url := fmt.Sprintf("%s%s", k8shost, "/.well-known/oauth-authorization-server")
+	req, err := http.NewRequest("GET", url, nil)
```

Comment:
While this is fine, JFI you could use ```requester.Get(url)``` to avoid having to build the request

Reply:
Yeah, I spotted that and thought about changing it. However, the ExternalHTTPRequester expects a Request object currently, so I decided to leave it for now, if that's OK?